rclpy/src/rclpy/_rclpy.c

```diff
@@ -3369,6 +3370,385 @@ rclpy_clock_set_ros_time_override(PyObject * Py_UNUSED(self), PyObject * args)
+      return false;
+    }
+    PyObject * parameter_dict;
+    if (!PyDict_Contains(node_params_dict, py_node_name)) {
+      parameter_dict = PyDict_New();
+      if (NULL == parameter_dict) {
+        Py_DECREF(py_node_name);
+        return false;
+      }
+      if (-1 == PyDict_SetItem(node_params_dict, py_node_name, parameter_dict)) {
+        Py_DECREF(py_node_name);
+        return false;
+      }
+    } else {
+      parameter_dict = PyDict_GetItem(node_params_dict, py_node_name);
+      if (NULL == parameter_dict) {
+        Py_DECREF(py_node_name);
+        return false;
+      }
+      /* This was a borrowed reference. INCREF'd so we can unconditionally DECREF below. */
+      Py_INCREF(parameter_dict);
+    }
+    rcl_node_params_t node_params = params->params[i];
+    for (size_t ii = 0; ii < node_params.num_params; ii++) {
+      PyObject * py_param_name = PyUnicode_FromString(node_params.parameter_names[ii]);
+      if (NULL == py_param_name) {
+        Py_DECREF(py_node_name);
+        Py_DECREF(parameter_dict);
+        return false;
+      }
+      PyObject * py_param = _parameter_from_rcl_variant(py_param_name,
+          &node_params.parameter_values[ii], parameter_cls, parameter_type_cls);
+      if (NULL == py_param) {
+        Py_DECREF(py_node_name);
+        Py_DECREF(parameter_dict);
+        Py_DECREF(py_param_name);
+        return false;
+      }
+      if (-1 == PyDict_SetItem(parameter_dict, py_param_name, py_param)) {
+        Py_DECREF(py_node_name);
+        Py_DECREF(py_param_name);
+        Py_DECREF(parameter_dict);
+        Py_DECREF(py_param);
+        return false;
+      }
+      Py_DECREF(py_param_name);
+      Py_DECREF(py_param);
+    }
+    Py_DECREF(py_node_name);
+    Py_DECREF(parameter_dict);
+  }
+  return true;
+}
+
+/// Populate a Python dict with node parameters parsed from arguments files
+/**
+ * On failure a Python exception is raised and false is returned if:
+ *
+ * Raises RuntimeError if param_files cannot be extracted from arguments.
+ * Raises RuntimeError if yaml files do not parse succesfully.
+ *
+ * \param[in] args The arguments to parse for parameter files
+ * \param[in] allocator Allocator to use for allocating and deallocating within the function.
+ * \param[out] params_by_node_name A Python dict object to place parsed parameters into.
+ *
+ * Returns true when parameters are parsed successfully (including the trivial case)
+ *         false when there was an error during parsing and a Python exception was raised.
+ *
+ */
+static bool
+_parse_param_files(
+  const rcl_arguments_t * args, rcl_allocator_t allocator, PyObject * parameter_cls,
+  PyObject * parameter_type_cls, PyObject * params_by_node_name)
+{
+  char ** param_files;
+  int param_files_count = rcl_arguments_get_param_files_count(args);
+  rcl_ret_t ret;
+  bool successful = true;
+  if (param_files_count > 0) {
+    ret = rcl_arguments_get_param_files(args, allocator, &param_files);
```

Comment:
as `ret` is not used anywhere else we could just check the return value of `rcl_arguments_get_param_files` and return on failure